src/Components/test/testassets/TestServer/Startup.cs

```diff
@@ -56,7 +56,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 
                 subdirApp.UseEndpoints(endpoints =>
                 {
-                    endpoints.MapHub<ComponentHub>(ComponentHub.DefaultPath).AddComponent<Index>(selector: "root");
+                    endpoints.MapHub<ComponentHub>(ComponentHub.DefaultPath).AddComponent<IEndpointConventionBuilder, Index>(selector: "root");
```

Comment:
We obviously don't want this.

Does `AddComponent` only make sense in the context of `MapHub`? Should `MapHub` return a custom builder type and `AddComponent` would then be an extension method of the custom builder type.

Reply:
yup.

Reply:
Which would you prefer:

* `public static IHubEndpointConventionBuilder AddComponent<TComponent>(this TBuilder builder, string selector)` - still has the generic type but would potentially lose the exact builder type
* `public static TBuilder AddComponent<TBuilder>(this TBuilder builder, string selector, Type component) where TBuilder : IHubEndpointConventionBuilder` - requires `typeof(Component)` but keeps the exact builder type

`IHubEndpointConventionBuilder` would be returned by https://github.com/aspnet/AspNetCore/blob/658b37d2bd3b0da2c07e25b53fa5c97ce8d656d3/src/SignalR/server/SignalR/src/HubEndpointRouteBuilderExtensions.cs#L23

Reply:
I don't think either one has strong downsides. 

`IHubConventionBuilder` is already strongly typed so 1 is not a major concern.
People like generics because they make them feel smart so 2 has that drawback. 

I would lean towards 1 because I don't think the drawback will matter in practice. I'd also encourage teh signelRz to return a *sealed class* from `MapHub<>`